src/components/capture/creation/DetailsForm.tsx

```diff
@@ -76,21 +73,26 @@ function NewCaptureDetails(props: NewCaptureDetailsProps) {
     };
 
     useEffect(() => {
-        if (connectors.length > 0) {
-            setSchema((previous: typeof schema) => {
-                previous.properties.image.oneOf = connectors.map(
-                    (connector) => {
-                        return {
-                            const: connector.links.images,
-                            title: connector.attributes.name,
-                        };
-                    }
-                );
+        if (isSuccess) {
+            if (connectorsData && connectorsData.data.length > 0) {
+                setSchema((previous: typeof schema) => {
+                    const listOfConnectors = connectorsData.data.map(
+                        (connector) => {
+                            return {
+                                const: connector.links.images,
+                                title: connector.attributes.name,
+                            };
+                        }
+                    );
+                    previous.properties.image.oneOf = listOfConnectors;
 
-                return previous;
-            });
+                    return previous;
+                });
+            }
+
+            setPostProcessingDone(true);
```

Comment:
This is where a form wants to do some "post processing" of the data before displaying itself. So we run some code and then set that state.